pkg/model/storage_spec.go

```diff
@@ -29,3 +29,12 @@ type StorageSpec struct {
 	// Additional properties specific to each driver
 	Metadata map[string]string `json:"Metadata,omitempty"`
 }
+
+// PublishedStorageSpec is a wrapper for a StorageSpec that has been published
+// by a compute provider - it keeps info about the hos, job and shard that
```

Comment:
Typo 😅

Reply:
But otherwise ++++++1 for this, StorageSpec has felt like a janky way of representing this for a long time

Reply:
lol - what even is a `hos` :-)